### ISSUE TITLE
add `dependsOn` property for nuxt plugins

### ISSUE THREAD

**Issue by huang-julien** [nuxt/nuxt#24123]
### Describe the feature

this feature would allow to make a plugin await hte end of the execution of another plugin before running it.

This can be a good improvement in multiple ways.

**encourage modules authors to set their plugins as parallel**
- In many case, using composition api implies that we'll expect to have some plugins to be ready before using it; So setting plugins as paarallel for module authors bring a sort of incertitude (for its users) of using a composition function which relies on a plugin (such as i18n)

**allow users to have a plugin which relies on multiple parallel plugins**  

**basically encourage users to set every plugins as parallel**

**reduce TTI**

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).
